server/api/controllers/webhook/getData.ts

```diff
@@ -0,0 +1,25 @@
+import { firestore } from './firebase';
+import { collection, query, getDocs } from '@firebase/firestore';
+const ref = collection(firestore, 'clients');
+type Record = { board_ID: string; username: string; api_key: string };
+const allData: Record[] = [];
+
+const getData = async () => {
+  const q = query(ref);
+
+  const querySnapshot = await getDocs(q);
+  querySnapshot.forEach((doc) => {
+    const data = doc.data();
+
+    if (data.board_ID && data.username && data.api_key) {
+      allData.push({
+        board_ID: data.board_ID,
+        username: data.username,
+        api_key: data.api_key,
+      });
+    }
+  });
+  return allData;
+};
+
+export default getData();
```

Comment:
This is not clear. You are returning the output of a function call that happens when you import that module? It's more complicated than it needs to be.

Just have this module return an async function, and the caller can call the function when they want the data.